### DISCUSSION TITLE
Pocketbase - modify Header

### DISCUSSION THREAD

**Discussion by jonathan-martz** [pocketbase/pocketbase#2618] [Q&A] [answered]
Iam currently working on a integration of Pocketbase into Shopware 6.

I habe the following Code:

const pb = new PocketBase('https://shop.restaurant-derschlag.de/store-api/pocketbase');
const result = await pb.collection('users').getList(1, 50);
console.log(result);

Is there a way to modify the Header of the Request ?
Shopware requires the:
Header "sw-access-key" is required.'

**Comment by ganigeorgiev** [comment, marked as answer]
Currently you can modify request headers only globally using the [`beforeSend` hook](https://github.com/pocketbase/js-sdk#send-hooks):

```js
pb.beforeSend = function (url, options) {
    // For list of the possible request options properties check
    // https://developer.mozilla.org/en-US/docs/Web/API/fetch#options
    options.headers['sw-access-key'] = 'test';

    return { url, options };
};
```

_There are plans in the near future to add support for custom headers and option to provide custom local `fetch` implementation per request basis._

**Reply by jonathan-martz** [reply]
That would be greate because Svelte uses a custom fetch function and send warnings if someone like pocketbase uses the default one.

**Comment by jonathan-martz** [comment]
Thanks that was the solution i was looking for.
